files/report.rb

```diff
@@ -121,6 +121,9 @@ def metrics_to_hash(report)
     end
     # fix for Puppet non-resource errors (i.e. failed catalog fetches before falling back to cache)
     report_status["failed"] += report.logs.find_all {|l| l.source =~ /Puppet$/ && l.level.to_s == 'err' }.count
+    if !SETTINGS.has_key?(:report_facter_errors) || SETTINGS[:report_facter_errors]
```

Comment:
You can use `fetch` to set a default when the key is missing.
```suggestion
    if SETTINGS.fetch(:report_facter_errors, true)
```